src/object_store/src/object/mod.rs

```diff
@@ -803,6 +820,16 @@ pub async fn parse_remote_object_store(
             .await
             .monitored(metrics),
         ),
+        hdfs if hdfs.starts_with("hdfs://") => {
+            let hdfs = hdfs.strip_prefix("hdfs://").unwrap();
+            let (_namenode, _root) = hdfs.split_once('@').unwrap();
+            ObjectStoreImpl::Opendal(
+                // TODO: create HDFS engine after HDFS env is is set up.
+                // OpendalObjectStore::new_hdfs_engine(namenode.to_string(), root.to_string())
+                //     .monitored(metrics),
```

Comment:
If we decided to use a feature flag to conrol which opendal engine is used, it is better to use conditional compilation `#[cfg(feature = "xxx")]` to control the behavior.